nightfall-optimist/src/utils/auth.mjs

```diff
@@ -0,0 +1,11 @@
+import config from 'config';
+import crypto from 'crypto';
+
+const environment = config.ENVIRONMENTS[process.env.ENVIRONMENT] || config.ENVIRONMENTS.localhost;
+
+export default function auth(req, res, next) {
+  const token = req.headers['x-app-token'];
+  if (token === crypto.createHash('sha256').update(environment.PROPOSER_KEY).digest('hex'))
```

Comment:
What happens if PROPOSER_KEY changes? Or there is no associated proposer key because this device is just a challenger? Or its undefined?
I think this is a source of problems. Change this to a fix value please

Reply:
If the key changes, you also change it on the HTTP header. I agree that if the device is just a challenger, there would not be a `PROPOSER_KEY` but a `CHALLENGER_KEY`, so I agree the naming should change to something like `PRIVATE_KEY`. As for being undefined, an empty key will fail, as the tests confirm.

We don't necessarily think this could be a source of problems, but we can pass yet another environment variable if you so prefer

Reply:
I do prefer having a separate env. My reason is that we won’t have a proposer always. The second, is that for usability during testing, i find it more convenient  to have the same token regardless of optimist were connecting

Reply:
Now addressed in #1255